src/math/src/bitmap.cairo

```diff
@@ -0,0 +1,270 @@
+// Internam imports
+
+use alexandria_math::fast_power::fast_power as pow;
+
+/// The bit value at the provided index of a number.
+/// # Arguments
+/// * `x` - The value for which to extract the bit value.
+/// * `i` - The index.
+/// # Returns
+/// * The value at index.
+pub fn get_bit_at<
```

Comment:
put all this in a trait like that 
https://github.com/starkware-libs/cairo/blob/de1d5b1a47864a67ac8e2432323944bccae5464c/corelib/src/serde.cairo#L90-L109